### ISSUE TITLE
associate()->with()

### ISSUE THREAD

**Issue by sebastianstoll97** [Crinsane/LaravelShoppingcart#240]
Great work, I must say.

I'm currently building a webshop CMS for myself, where I'm running this setup: Products hasMany Variants, and it's then the variants that is added to the cart.

Therefore, it'd be awesome(!!) if it was possible to run a with() statement on the associate() method? :-)

**Comment by Crinsane** [comment]
Can't you just call `$cartItem->model->variants`?

**Comment by eberkund** [comment]
I don't know if this any help to you but in my shop each CartItem has a Product model and another model specific to the user buying it associated with it. The product model is what implements the Buyable interface so for the additional model I just reference its ID in the options parameter. 

    Cart::add($product, $quantity, [
        'other_model' => $id
    ])

**Comment by sebastianstoll97** [comment]
@Crinsane Yes, also I do. When adding to card, I add the variant (of course), but when I display the cart, it'd be nice if I could get the Product too, because it have some properties that would be nice to show. :-)

Nice one, @eberkund ! :) I could use that.

**Comment by sebastianstoll97** [comment]
I've built this helper function, which actually solves my needs. It works for now :)
https://gist.github.com/sebastianstoll97/471e0c3d531a6cd2f5de4e9df86eb431

**Comment by IrsyadAdl** [comment]
This is not working.
> Can't you just call $cartItem->model->variants?

I have made like the read me file
```php
        $cartItem = Cart::add($product->id, $product->title, $quantity, $product->price, ['stock' => $product->stock]);
        Cart::associate($cartItem->rowId, 'App\Product');
```
